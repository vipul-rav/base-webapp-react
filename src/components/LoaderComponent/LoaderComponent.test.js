import React from 'react';
import { render } from 'setupTests';
import { LoaderComponent } from './LoaderComponent';
import configureMockStore from 'redux-mock-store';

const content = {
    ExitText: 'Exit',
    btnContinue: 'Continue',
    homePageText: 'Home Page',
};

const setup = overridesProps => {
    const props = {
        content: content,

        ...overridesProps,
    };

    const mockStore = configureMockStore();
    const store = mockStore({
        config: {
            externalContent: content,
        },
    });

    const { wrapper, getByText } = render(
        <LoaderComponent {...props} store={store} />
    );

    return {
        props,
        wrapper,
        getByText,
    };
};

describe('Loader Component', () => {
    it('should render', () => {
        const overrideProps = {
            isLoading: true,
            content: {
                loadingText: 'Please wait...',
            },
        };
        const { wrapper } = setup(overrideProps);
        expect(wrapper).toMatchSnapshot();
    });
});
