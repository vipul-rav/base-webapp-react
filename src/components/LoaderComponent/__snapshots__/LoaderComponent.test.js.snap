// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader Component should render 1`] = `undefined`;
